src/utils.ts

```diff
@@ -33,9 +33,29 @@ async function normalizeStore(source: string | ZarrArray['store']) {
 
 export async function open(source: string | ZarrArray['store']) {
   const { store, path } = await normalizeStore(source);
-  return openGroup(store, path).catch((err) => {
+
+  function nested(store) {
+    const get = (target, key) => {
+      if (key === 'getItem' || key === 'setItem' || key === 'containsItem') {
+        return (path, ...args) => {
+          if (path.endsWith('.zarray') || path.endsWith('.zattrs') || path.endsWith('.zgroup')) {
+            return target[key](path, ...args);
+          }
+          const prefix = path.split('/');
+          const chunkKey = prefix.pop();
+          const newPath = [...prefix, chunkKey.replaceAll('.', '/')].join('/');
+          return target[key](newPath, ...args);
+        }
+      }
+      return Reflect.get(target, key);
+    };
+    return new Proxy(store, { get });
+  }
+  const nested_store = nested(store);
```

Comment:
I would actually move this to be a utility function (outside of `open`), since this would break vizarr for folks who don't have a nested store (e.g. imjoy usage, Jupyter notebooks). 

My thought is that in `io.ts`, we can check for OME-Zarr version and recreate the group if needed:

```typescript
if (node instanceof ZarrGroup) {
    const attrs = (await node.attrs.asObject()) as Ome.Attrs;

   // Not sure how to make this check
   if (/* attrs.version === 2 */) {
       // just mutate the store here
       node.store = nested(node.store);
    }

    if ('plate' in attrs) {
      return loadPlate(config, node, attrs.plate);
    }

    if ('well' in attrs) {
      return loadWell(config, node, attrs.well);
    }

/* ... */
```